backend/core/src/main/java/org/sonarsource/sonarlint/core/fs/FileExclusionService.java

```diff
@@ -314,6 +325,37 @@ private void logFilteredURIs(String reason, ArrayList<URI> uris) {
     }
   }
 
+  private static boolean isLargerThanMaxAutoSize(ClientFile file) {
```

Comment:
You use many methods from `ClientFile`, this looks like a 'feature envy' code smell. I think you could move most of this to `ClientFile` with a `ClientFile#isLargerThan(long size)`